src/iface/fragmentation.rs

```diff
@@ -3,28 +3,28 @@
 use managed::{ManagedMap, ManagedSlice};
 
 use crate::storage::Assembler;
-use crate::time::Instant;
+use crate::time::{Duration, Instant};
 use crate::Error;
 use crate::Result;
 
 /// Holds different fragments of one packet, used for assembling fragmented packets.
 #[derive(Debug)]
-#[cfg_attr(feature = "defmt", derive(defmt::Format))]
 pub struct PacketAssembler<'a> {
     buffer: ManagedSlice<'a, u8>,
     assembler: AssemblerState,
 }
 
 /// Holds the state of the assembling of one packet.
 #[derive(Debug, PartialEq)]
-#[cfg_attr(feature = "defmt", derive(defmt::Format))]
 enum AssemblerState {
     NotInit,
+    Discarded,
```

Comment:
Why is the new `Discarded` state needed? It adds quite a lot of complexity, I think it should be possible to do ipv4 assembling without it.

Reply:
So I was playing with the ICMP Time Exceeded message. And I had to store the first 8 bytes somewhere. That somewhere was in the Discarded state, because the `mark_discarded_when` function checked if the first 8 bytes were already received and stored them in `Discarded`. Previous version had `Option<[u8;8]>`. But then it became a real mess to deal with everything and this is just a leftover of me playing with that. But indeed, if we don't do the ICMP thing, then I can remove it.